Contents/Items/Card/CardUI.cs

```diff
@@ -0,0 +1,332 @@
+﻿using System;
+using Terraria;
+using Terraria.UI;
+using ReLogic.Content;
+using Terraria.ModLoader;
+using Microsoft.Xna.Framework;
+using System.Collections.Generic;
+using BossRush.Contents.Artifacts;
+using Terraria.GameContent.UI.Elements;
+using Microsoft.Xna.Framework.Graphics;
+using Terraria.GameContent;
+using BossRush.Contents.WeaponModification;
+using System.Linq;
+using BossRush.Contents.Perks;
+
+namespace BossRush.Contents.Items.Card {
+	internal class CardUI : UIState {
+		public List<PlayerStats> CardStats = new List<PlayerStats>();
+		public List<float> CardStatsNumber = new List<float>();
+		public float Multiplier = 1f;
+		/// <summary>
+		/// 1 = Copper<br/>
+		/// 2 = Silver<br/>
+		/// 3 = Gold<br/>
+		/// 4 = Platinum<br/>
+		/// </summary>
+		public int Tier = 0;
+		public int PostTierModify = 0;
+		public int CursedID = -1;
+		public override void OnActivate() {
+			Elements.Clear();
+			Player player = Main.LocalPlayer;
+			if (player.TryGetModPlayer(out PlayerCardHandle modplayer)) {
+				if (Tier <= 0)
+					return;
+				bool hasMagicDeck = player.HasArtifact<MagicalCardDeckArtifact>();
+				PostTierModify = hasMagicDeck ? Tier + 1 : Tier;
+				for (int l = 0; l < 3; l++) {
+					SetBadStatsBaseOnTier(modplayer, hasMagicDeck);
+					int offset = 0;
+					if (CardStats.Count > 0)
+						offset++;
+					int cardlength = Math.Clamp(PostTierModify + offset, 0, 19);
+					AddCardStatsAndValue(modplayer, offset, cardlength);
+					CursedHandle(modplayer);
+					CardStatsIncreasesSelection cardUI = new CardStatsIncreasesSelection(TextureAssets.InventoryBack10, modplayer);
+					cardUI.CardStats.AddRange(CardStats);
+					cardUI.CardStatsNumber.AddRange(CardStatsNumber);
+					cardUI.PostTierModify = PostTierModify;
+					cardUI.UISetPosition(player.Center - new Vector2(-100 + 100 * l, 40), new Vector2(26, 26));
+					cardUI.UISetWidthHeight(52, 52);
+					Append(cardUI);
+					CardStats.Clear();
+					CardStatsNumber.Clear();
+					CursedID = -1;
+				}
+				ImprovisedUITextBox itemtextbox = new ImprovisedUITextBox("");
+				itemtextbox.SetTextMaxLength(1000);
+				string lines = "";
+				itemtextbox.ImprovisedUIpanel_Height= 80;
+				itemtextbox.SetText(lines);
+				itemtextbox.Recalculate();
+				itemtextbox.IgnoresMouseInteraction = true;
+				itemtextbox.ShowInputTicker = false;
+				itemtextbox.UISetPosition(player.Center + new Vector2(0, 100), new Vector2(26, 26));
+				Append(itemtextbox);
+			}
+		}
+		private void CursedHandle(PlayerCardHandle modplayer) {
+			if (CursedID != -1) {
+				List<int> list = new List<int> { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 };
+				if (modplayer.listCursesID.Count < list.Count) {
+					foreach (var item in modplayer.listCursesID) {
+						if (list.Contains(item))
+							list.Remove(item);
+					}
+					if (list.Count > 1)
+						CursedID = Main.rand.Next(list);
+					else
+						CursedID = list[0];
```

Comment:
Could simplify to just `CursedID = Main.rand.Next(list);`, since behaviour-wise it's same thing.